feat: [PKD-002] Pokemon Information Page: Base Stats

Enrickzz opened the pull request:
Redesigned pokemon_detailed_page.
Mapped data in DetailsPokemon.
DetailsPage VM and connector
Stats tab widget
Loading Widgets

Enrickzz committed:
feat: [PKD-002] Pokemon Information Page: Base Stats

Redesigned pokemon_detailed_page.
Mapped data in DetailsPokemon.
DetailsPage VM and connector
Loading Widgets

Enrickzz commented:
https://user-images.githubusercontent.com/61441278/180945306-a58f53f8-4b78-4ea1-9e6a-c01ff3de8454.mp4

Short demo

dbenedictos commented on lib/api/models/pokemon_type_model.dart:
```diff
@@ -8,7 +8,7 @@ part 'pokemon_type_model.g.dart';
   factory PokemonType({
     @JsonKey(name: 'id') int? id,
-    @JsonKey(name: 'types') List<SubType>? types,
+    @JsonKey(name: 'types') List<SubType>? subType,
```
please make the name property inside the JsonKey identical to your variable name

dbenedictos replied:
please rename the name parameter to subTypes

dbenedictos commented on lib/api/models/sub_type_model.dart:
```diff
@@ -8,7 +8,7 @@ part 'sub_type_model.g.dart';
   factory SubType({
     @JsonKey(name: 'slot') int? slot,
-    @JsonKey(name: 'type') SpecificType? type,
+    @JsonKey(name: 'type') SpecificType? specificType,
```
same here

dbenedictos replied:
@Enrickzz

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -1,24 +1,176 @@
-              style: TextStyle(color: Colors.black),
-            ),
-            backgroundColor: Colors.transparent,
+            backgroundColor: Colors.white, //make dynamic laer
```
proper todos. ex:
`// TODO: Make dynamic later`

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -1,24 +1,176 @@
+          SliverFillRemaining(
+            hasScrollBody: false,
+            child: Container(
+                color: Colors.white, // make dynamic later
```
same as above

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -1,24 +1,176 @@
+                                children: [
+                                  Container(
+                                    decoration: BoxDecoration(
+                                      color: Colors.grey, // change color later
```
and here :P

dbenedictos commented on lib/state/actions/home_page_actions.dart:
```diff
@@ -17,3 +18,13 @@ class GetPokemonList extends LoadingAction {
     );
   }
 }
+class GetPokemonDetails extends LoadingAction {
```
format this file

Enrickzz replied:
already formatted

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page.dart:
```diff
@@ -1,24 +1,176 @@
+                        child: Image.network(
+                          '$pokemonImgUrl${pokemon?.id}.png',
+                          fit: BoxFit.cover,
+                          errorBuilder: (context, error, stackTrace) {
```
for callbacks that contains parameters that we don't use, assign it to `_`. so in this case `(_, __, ___)`

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_base_stats_tab.dart:
```diff
@@ -0,0 +1,50 @@
+
+class BaseStats extends StatelessWidget {
+  const BaseStats({required this.stat});
+  final List<Stat>? stat;
```
since your constructor requires this. Please remove the null check

dbenedictos commented on lib/feature/pokemon_details_page.dart/widgets/pokemon_base_stats_tab.dart:
```diff
@@ -0,0 +1,50 @@
+                      color: Colors.black,
+                      backgroundColor: Colors.grey,
+                      valueColor: AlwaysStoppedAnimation<Color>(Colors.black),
+                      value: dblStat! / 100,
```
I know pokemons can have more than 100 as there stat value

dbenedictos commented on lib/feature/widgets/loading_widgets.dart:
```diff
@@ -0,0 +1,70 @@
+      appBar: AppBar(
+        leading: BackButton(color: Colors.black),
+        title: const Text(
+          'Pokédex',
```
strings file

dbenedictos commented on lib/feature/widgets/loading_widgets.dart:
```diff
@@ -0,0 +1,70 @@
+  }
+}
+
+class PokemonListLoading extends StatelessWidget {
```
separate file

dbenedictos commented on lib/feature/widgets/loading_widgets.dart:
```diff
@@ -0,0 +1,70 @@
+}
+
+class PokemonListLoading extends StatelessWidget {
+  const PokemonListLoading();
```
and this widget is not used

dbenedictos commented on lib/feature/pokemon_details_page.dart/pokemon_details_page_connector.dart:
```diff
@@ -0,0 +1,26 @@
+
+class PokemonDetailsConnector extends StatelessWidget {
+  const PokemonDetailsConnector({this.url});
+  final url;
```
variable type

dbenedictos commented on lib/state/actions/home_page_actions.dart:
```diff
@@ -17,3 +18,13 @@ class GetPokemonList extends LoadingAction {
+class GetPokemonDetails extends LoadingAction {
+  static const key = 'get_pokemon_details_key';
+  GetPokemonDetails(this.url) : super(actionKey: key);
+  final url;
```
variable type

Enrickzz committed:
PR Feedback

- Variable name changes (sub_type_model)
- Added constants in doubles and strings constant files
- Separated loading widgets files
- Removed null checker (List<Stat>?)
- Revised TODO: comments

dbenedictos commented on lib/feature/home_page/widgets/pokemon_tile.dart:
```diff
@@ -2,6 +2,7 @@ import 'dart:convert';
 import 'package:pokedex_async_redux/api/models/pokemon_model.dart';
 import 'package:pokedex_async_redux/api/models/pokemon_type_model.dart';
 import 'package:pokedex_async_redux/feature/pokemon_details_page.dart/pokemon_details_page.dart';
```
unused import

Enrickzz committed:
PR Feedback

- Renamed subType => subType
- Updated DetailedPokemonHandler
- Updated PokemonTile
- Minor DetailsPage UI changes
- Assigned variable types

dbenedictos commented on lib/feature/home_page/widgets/pokemon_tile.dart:
```diff
@@ -23,8 +25,19 @@ class _PokemonTileState extends State<PokemonTile> {
         var result = jsonDecode(value.body);
+
+        List SubTypeMap = result['types'];
+        print(SubTypeMap);
```
remove, haha

Enrickzz committed:
PR Feedback

- Remove unnecessary print()
